MBBSEmu/CPU/CPUCore.cs

```diff
@@ -2260,68 +2320,63 @@ private void Op_Call()
         /// <summary>
         ///     Floating Point Load Operation (x87)
         /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [MethodImpl(CompilerOptimizations)]
         private void Op_Fld()
         {
-            var offset = GetOperandOffset(_currentInstruction.Op0Kind);
-            var floatToLoad = Memory.GetArray(Registers.DS, offset, 4);
-            FpuStack[Registers.Fpu.GetStackTop()] = floatToLoad.ToArray();
+            var floatToLoad = GetOperandValueFloat(_currentInstruction.Op0Kind, EnumOperandType.Source);
+
+            //Set Value as new ST(0)
             Registers.Fpu.PushStackTop();
+            FpuStack[Registers.Fpu.GetStackTop()] = floatToLoad;
         }
 
         /// <summary>
         ///     Integer Load Operation (x87)
         /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [MethodImpl(CompilerOptimizations)]
         private void Op_Fild()
         {
-            var offset = GetOperandOffset(_currentInstruction.Op0Kind);
-            var intToLoad = Memory.GetArray(Registers.DS, offset, 2);
+            var intToLoad = GetOperandValueUInt16(_currentInstruction.Op0Kind, EnumOperandType.Source);
 
-            var convertedInt = Convert.ToSingle(BitConverter.ToInt16(intToLoad));
+            var convertedInt = Convert.ToSingle(intToLoad);
 
-            FpuStack[Registers.Fpu.GetStackTop()] = BitConverter.GetBytes(convertedInt);
             Registers.Fpu.PushStackTop();
+            FpuStack[Registers.Fpu.GetStackTop()] = convertedInt;
+
         }
 
         /// <summary>
         ///     Floating Point Multiplication (x87)
         /// </summary>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [MethodImpl(CompilerOptimizations)]
         private void Op_Fmul()
         {
-            var offset = GetOperandOffset(_currentInstruction.Op0Kind);
-            var floatToMultiply = Memory.GetArray(Registers.DS, offset, 4);
-
-            Registers.Fpu.PopStackTop();
-            var float1 = BitConverter.ToSingle(FpuStack[Registers.Fpu.GetStackTop()]);
-            var float2 = BitConverter.ToSingle(floatToMultiply);
+            var ST0 = FpuStack[Registers.Fpu.GetStackTop()];
+            var source = GetOperandValueFloat(_currentInstruction.Op0Kind, EnumOperandType.Source);
 
-            var result = float1 * float2;
-            FpuStack[Registers.Fpu.GetStackTop()] = BitConverter.GetBytes(result);
-            Registers.Fpu.PushStackTop();
+            var result = ST0 * source;
```

Comment:
do these need to be wrapped in unchecked?

Reply:
You know, I had these there because I thought it wouldn't gracefully handle the overflow but it seems to handle it differently than integer types where it just casts it as NaN. I went ahead and removed it from the tests.